themes/steeef.zsh-theme

```diff
@@ -62,7 +62,7 @@ zstyle ':vcs_info:*:prompt:*' nvcsformats   ""
 
 
 function steeef_preexec {
-    case "$(history $HISTCMD)" in
+    case "$(whence -c $(fc -l $HISTCMD $HISTCMD))" in
```

Comment:
Use `whence` without the `-c`, otherwise you'll get the verbose output `name: aliased to command`.
Also, we can use `$1` since this is a preexec function, which contains the command about to be executed.

Thirdly, we want to just `whence` the first part of the command. Otherwise we might whence an argument instead that contains the word git, which is not what we want. For that we can split the command by spaces  and get the first part.

All together:
```
local -a cmd=(${(s: :)1})
case "$(whence $cmd[1])" in
....
```

For example:
```
$ 1="gp origin"; cmd=(${(s: :)1}); whence $cmd[1]
git push
```

Reply:
Oh, even better. On a preexec function `$1` gives you the typed command, `$2` gives you the expanded command, so we don't need `ẁhence`. Just replace the whole thing with
```zsh
case "$2" in
```

Cool huh?

Reply:
That's great!